### ISSUE TITLE
[java] ConstantsInInterface false negative suppressed by name collision in nested enum members

### ISSUE THREAD

**Issue by leemeii** [pmd/pmd#6298]
**Affects PMD Version:** 6.51.0

**Rule:** ConstantsInInterface 
[https://docs.pmd-code.org/latest/pmd_rules_java_design.html#constantsininterface](https://www.google.com/search?q=https://docs.pmd-code.org/latest/pmd_rules_java_design.html%23constantsininterface)

**Description:**
I have discovered a False Negative in the ConstantsInInterface rule.

The rule fails to report a violation for a constant defined in an interface if that interface contains a nested enum which has a field and a method with the same name.

It seems that the presence of this specific structure (name collision between a field and a method within the inner enum) interferes with the rule's detection logic for the outer interface, causing it to miss obvious violations. If the conflicting members in the inner enum are removed, the rule reports the violation correctly.

**Code Sample demonstrating the issue:**

```java
public interface RoadFeature {

    String STOP = "Stop"; // PMD SHOULD report a violation here, but it doesn't.

    enum Lights {
        RED,
        YELLOW,
        GREEN;

        // The presence of this field 'bar' and method 'bar()' seems to suppress the rule for the outer interface.
        private int bar = 0;

        public void bar() {
           System.out.println("This is a method named bar");
        }
    }
}
```

**Expected outcome:**

PMD should report a violation at line 3 (String STOP = "Stop";), but doesn't. This is a false-negative.

**Running PMD through:** CLI

**Comment by mrclmh** [comment]
In my tests this is reproducible in 6.55.0 but not in 7.0.0 or 7.19.0.

**Comment by leemeii** [comment]
Thanks for the update @mrclmh.

I have verified this locally with **PMD 7.19.0 (CLI)**.

You are right, the specific masking effect described in this issue is **no longer reproducible**.

However, I noticed something interesting: even with a minimal baseline case (an interface containing ONLY a constant, `String STOP = "Stop";`), PMD 7.19.0 reports **0 violations** using `category/java/design.xml`.

It seems the `ConstantsInInterface` rule behavior has changed significantly or is less strict by default in PMD 7.

Since the original issue (masking by inner enum) is definitely gone (or moot due to the rule behavior change), I will close this issue.

Thanks for your help!

**Comment by mrclmh** [comment]
The rule is currently in the [category/java/bestpractices.xml](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#constantsininterface) category, so it can not be reported in the `design` category.

**Comment by leemeii** [comment]
@mrclmh You are absolutely right, thank you for pointing that out!

I was testing against `category/java/design.xml` based on PMD 6 behavior, which caused the false assumption that the rule was ignored.

After switching to **`category/java/bestpractices.xml`**, I verified that PMD 7.19.0 correctly reports violations for **BOTH** the baseline case and the triggering case (with the nested enum/method collision).
